tests/D2L.CodeStyle.Analyzers.Test/Specs/ConstantAnalyzer.cs

```diff
@@ -0,0 +1,69 @@
+﻿// analyzer: D2L.CodeStyle.Analyzers.Helpers.ConstantAnalyzer
+
+using D2L.CodeStyle.Annotations;
+
+#region Relevant Types
+namespace D2L.CodeStyle.Annotations
+{
+	public static class Contract
+	{
+		public sealed class ConstantAttribute : Attribute { }
+	}
+}
+#endregion
+
+namespace SpecTests
+{
+	using static Contract;
+
+	public sealed class Types
+	{
+		public static void SomeMethodWithConstantParameter([Constant] string param1) { }
+		public static void SomeMethodWithParameter(string param1) { }
+		public static void SomeMethodWithOneConstantParameter([Constant] string param1, string param2) { }
+		public static void SomeMethodWithOneOtherConstantParameter(string param1, [Constant] string param2) { }
+		public static void SomeMethodWithTwoConstantParameters([Constant] string param1, [Constant] string param2) { }
+	}
+
+	public sealed class Tests
+	{
+		void Method()
+		{
+			string variable = "This is a variable message";

```

Comment:
Most non-constant usage of the logger look like

```csharp
m_logger.Error( string.Format( ERROR_RELATIVE_MONTH_DATE_VALUE_INVALID, achievementId ) );
m_logger.Warn( $"The job {jobId} for the org unit {entity.TargetOrgUnitId} has/is being proccessed" );
```

I would personally include tests that cover those cases.